src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5356,6 +5369,12 @@ private bool DetermineIfDefinitelyTypeArgumentList(bool isDefinitelyTypeArgument
             return isDefinitelyTypeArgumentList;
         }
 
+        private bool isPossibleTypeInTypeArgumentList()
+        {
+            return this.IsPossibleType()
+                && !((_termState & TerminatorState.IsEndOfReturnType) == TerminatorState.IsEndOfReturnType && this.IsEndOfReturnType());
```

Comment:
i find this entire expressoin very hard to read.  Can we just break this up into simple statements that explain what they're trying to check for?  Terseness is not a virtue here :)